# HG changeset patch
# User Rik <rik@octave.org>
# Date 1376507151 25200
#      Wed Aug 14 12:05:51 2013 -0700
# Node ID dac81d4b8ce16d7565a4113765404ffbe3c11a26
# Parent  684ccccbc15d8dfcc1b294f5d389f52097a78b54
assert.m: Fix typo stopping comparison of imaginary non-infinite values.

* scripts/testfun/assert.m: Fix typo stopping comparison of imaginary
non-infinite values.

diff --git a/scripts/testfun/assert.m b/scripts/testfun/assert.m
--- a/scripts/testfun/assert.m
+++ b/scripts/testfun/assert.m
@@ -265,17 +265,17 @@ function assert (cond, varargin)
         ## Replace exceptional values already checked above by zero.
         A_null_real = real (A);
         B_null_real = real (B);
         exclude = ! isfinite (A_null_real) & ! isfinite (B_null_real);
         A_null_real(exclude) = 0;
         B_null_real(exclude) = 0;
         A_null_imag = imag (A);
         B_null_imag = imag (B);
-        exclude = ! isfinite (A_null_real) & ! isfinite (B_null_real);
+        exclude = ! isfinite (A_null_imag) & ! isfinite (B_null_imag);
         A_null_imag(exclude) = 0;
         B_null_imag(exclude) = 0;
         A_null = complex (A_null_real, A_null_imag);
         B_null = complex (B_null_real, B_null_imag);
         if (isscalar (tol))
           mtol = repmat (tol, size (A));
         else
           mtol = tol;
